File: examples/authentication/roles/PingCommand.java
package pw.mihou.nexus.commands;

import pw.mihou.nexus.core.reflective.annotations.Share;
import pw.mihou.nexus.features.command.facade.NexusCommandEvent;
import pw.mihou.nexus.features.command.facade.NexusHandler;
import pw.mihou.nexus.features.command.interceptors.commons.NexusCommonInterceptors;

import java.util.List;

public class PingCommand implements NexusHandler {

    private final String name = "ping";
    private final String description = "An example of how a command can share custom fields to a middleware, etc.";

    private final List<String> middlewares = List.of(NexusCommonInterceptors.NEXUS_AUTH_ROLES_MIDDLEWARE);
    @Share private final List<Long> requiredRoles = List.of(949964174505689149L);

    @Override
    public void onEvent(NexusCommandEvent event) {
        event.respondNowWith("Pong!");
    }
}

File: examples/router/PingCommand.kt
package pw.mihou.nexus.features.command.router.example

import org.javacord.api.interaction.SlashCommandOption
import org.javacord.api.interaction.SlashCommandOptionType
import pw.mihou.nexus.features.command.facade.NexusCommandEvent
import pw.mihou.nexus.features.command.facade.NexusHandler
import pw.mihou.nexus.features.command.router.SubcommandRouter

object PingCommand: NexusHandler {
    private val name = "ping"
    private val description = "Pings and pongs!"

    private val options = listOf(
        SlashCommandOption.create(SlashCommandOptionType.SUB_COMMAND, "pong", "Performs a pong!"),
        SlashCommandOption.create(SlashCommandOptionType.SUB_COMMAND, "ping", "Performs a ping!")
    )

    private val router = SubcommandRouter.create {
        route("ping", PingSubcommand)
        route("pong", PongSubcommand)
    }

    override fun onEvent(event: NexusCommandEvent) = router.accept(event) {
        // Optionally you can add additional shared data such as this:
        event.store("message", "Ping pong! You can add stuff like this here!")
    }
}